File: php/testcode/pathtraver_002.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_pt_basename_validated
function pathtraver_basename_validated(BenchmarkRequest $req): BenchmarkResponse {
    $file = $req->param('file');
    $safe = basename($file);
    $content = file_get_contents("/uploads/" . $safe); // vuln-code-snippet safe-line php_pt_basename_validated
    return BenchmarkResponse::ok($content);
}
// vuln-code-snippet end php_pt_basename_validated

File: ruby/testcode/pathtraver_002.rb
require_relative 'shared'

# vuln-code-snippet start ruby_pt_basename
def get_uploaded_file_safe(req)
  filename = req.param('filename')
  safe = File.basename(filename)
  content = File.read("/uploads/" + safe) # vuln-code-snippet safe-line ruby_pt_basename
  BenchmarkResponse.ok(content)
end
# vuln-code-snippet end ruby_pt_basename
